### PR TITLE
[yamaha] resolved cyclic dependency

### PR DESCRIPTION
**Pull request by sjka**
...between the ChannelTypeProvider and the ChannelTypeRegistry
which was established well-hidden via the TypeResolver.

Signed-off-by: Simon Kaufmann <simon.kfm@googlemail.com>

### CODE REVIEW

**Comment by sjka** [comment]
hang on, that's only half of it, I just realized...

**Comment by sjka** [comment]
That should be it now

File: addons/binding/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/internal/ChannelsTypeProviderAvailableInputs.java [hunk 1]
```diff
@@ -72,35 +68,63 @@ public ChannelTypeUID getChannelTypeUID() {
...
-        if (templateType == null) {
-            logger.warn("Couldn't find: {}", channelTypeInputsUID);
-        } else {
-            channelType = new ChannelType(channelTypeUID, false, "String", templateType.getKind(),
-                    templateType.getLabel(), templateType.getDescription(), templateType.getCategory(),
-                    templateType.getTags(), templateType.getState(), templateType.getEvent(),
-                    templateType.getConfigDescriptionURI());
-        }
+    private void createChannelType(StateDescription state) {
+        channelType = new ChannelType(channelTypeUID, false, "String", ChannelKind.STATE, "Input source",
```
Comment (davidgraeff) [inline addons/binding/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/internal/ChannelsTypeProviderAvailableInputs.java]: Are those strings (and the StateOption("...", "Net Radio") ones) still translatable? I like the xml-less solution.

Comment (martinvw) [reply addons/binding/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/internal/ChannelsTypeProviderAvailableInputs.java]: @sjka wdyt?

> Are those strings (and the StateOption("...", "Net Radio") ones) still translatable? I like the xml-less solution.

Comment (martinvw) [reply addons/binding/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/internal/ChannelsTypeProviderAvailableInputs.java]: @sjka can you respond to @davidgraeff his question.

**Pull request merged**